resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -583,6 +632,9 @@ private[spark] class Client(
       case _ => None
     }
 
+    // If preload is enabled, preload the statCache with the files in the directories
+    if (statCachePreloadedEnabled) { statCachePreload(fs, statCache) }
```

Comment:
The control flow is a little confusing to me, where we initialize `statCache` up on L537, then access it while processing the Ivy settings and AM keytab file, _then_ populate it down here. How about we preload it when we initialize?
```scala
val statCache = if (statCachePreloadEnabled) {
  getPreloadedStatCache(fs)
} else {
  HashMap[URI, FileStatus]()
}
```